rle/runs.go

```diff
@@ -204,3 +204,32 @@ func (ri *RunSliceIterator) NextRun() (Run, error) {
 	ri.i++
 	return out, nil
 }
+
+type notIter struct {
+	it RunIterator
+}
+
+func (ni *notIter) HasNext() bool {
+	return true
```

Comment:
I think `HasNext` always returning true might lead to trouble, or maybe not.

Reply:
Well, in the usecase I have for it, its fine. Ideally we can mark runs as being of infinite length, and things can handle that specially